fastscanner/services/indicators/lib/candle.py

```diff
@@ -273,58 +336,51 @@ async def extend(self, symbol: str, df: pd.DataFrame) -> pd.DataFrame:
             df[self.column_name()] = pd.NA
             return df
 
-        daily_df = (
-            daily_df[[CandleCol.HIGH, CandleCol.LOW]]
+        rolling_df = (
+            daily_df[[self._candle_col]]
             .rolling(self._n_days, min_periods=1)
-            .agg(
-                {
-                    CandleCol.HIGH: "max",
-                    CandleCol.LOW: "min",
-                }
-            )
-            .rename(
-                columns={
-                    CandleCol.HIGH: "_highest",
-                    CandleCol.LOW: "_lowest",
-                }
-            )
+            .agg(self._operation)
+            .rename(columns={self._candle_col: f"_rolling_{self._candle_col}"})
             .set_index(daily_df.index.date)  # type: ignore
         )
 
-        daily_df.loc[df.index[-1].date(), ["_highest", "_lowest"]] = pd.NA
-        daily_df = daily_df.shift(1)
+        rolling_df.loc[df.index[-1].date(), f"_rolling_{self._candle_col}"] = pd.NA
 
+        rolling_df = rolling_df.shift(1)
         df.loc[:, "date"] = df.index.date  # type: ignore
-        df = df.join(daily_df, on="date")
-        df[self.column_name()] = (df[CandleCol.CLOSE] - df.loc[:, "_lowest"]) / (
-            df.loc[:, "_highest"] - df.loc[:, "_lowest"]
-        )
+        df = df.join(rolling_df, on="date")
+
+        df[self.column_name()] = df[f"_rolling_{self._candle_col}"]
 
-        return df.drop(columns=["date", "_lowest", "_highest"])
+        return df.drop(columns=["date", f"_rolling_{self._candle_col}"])
 
     async def extend_realtime(self, symbol: str, new_row: pd.Series) -> pd.Series:
         assert isinstance(new_row.name, datetime)
         last_date = self._last_date.get(symbol)
+
         if last_date is None or last_date != new_row.name.date():
             daily_df = await self._get_high_low_n_days(symbol, new_row.to_frame().T)
-            self._last_date[symbol] = new_row.name.date()  # type: ignore
-            self._low_n_days[symbol] = daily_df[CandleCol.LOW].to_list()[
-                -self._n_days :
-            ]
-            self._high_n_days[symbol] = daily_df[CandleCol.HIGH].to_list()[
+            self._last_date[symbol] = new_row.name.date()
+
+            self._rolling_values[symbol] = daily_df[self._candle_col].to_list()[
                 -self._n_days :
             ]
-        if len(self._high_n_days.get(symbol, [])) == 0:
-            logger.debug(f"{symbol}] Insufficient high/low data for indicator")
-            return new_row
 
-        last_high = max(self._high_n_days[symbol])
-        last_low = min(self._low_n_days[symbol])
-        last_close = new_row[CandleCol.CLOSE]
-        range_val = last_high - last_low
-        if range_val == 0:
+        values = self._rolling_values.get(symbol, [])
+
+        if not values:
+            logger.debug(f"{symbol}] Insufficient data for {self.column_name()}")
```

Comment:
There is another log above that warns about missing data. We don't need this one